applications/ParticleMechanicsApplication/custom_strategies/strategies/MPM_strategy.h

```diff
@@ -461,61 +468,210 @@ class MPMStrategy
         mp_solving_strategy->SetEchoLevel(Level);
     }
 
-    //*********************************************************************************
-    /**OPERATIONS ACCESSIBLE FROM THE INPUT:*/
-    /*@{ */
-
-    /**
-    operation to predict the solution ... if it is not called a trivial predictor is used in which the
-    values of the solution step of interest are assumed equal to the old values
-     */
-    void Predict() override
-    {
-    }
-
     /**
-    Initialization of member variables and prior operations
+     * @brief Initialization of member variables and prior operations
      */
     void Initialize() override
     {
     }
 
     /**
-    the problem of interest is solved
+     * @brief Performs all the required operations that should be done (for each step) before solving the solution step.
      */
-    double Solve() override
+    void InitializeSolutionStep() override
     {
-        // Check which nodes and elements are ACTIVE and populate the MPM model part
-        KRATOS_INFO_IF("MPM_Strategy", this->GetEchoLevel() > 1) << "Main Solve - Start" <<std::endl;
-        KRATOS_INFO_IF("MPM_Strategy", this->GetEchoLevel() > 1) << "Search Element - Start" <<std::endl;
-        this->SearchElement(mr_grid_model_part, mr_mpm_model_part);
-
-        // Only perform this once
+        // Initialize solving strategy: only to be done at the beginning of time step
         mp_solving_strategy->Initialize();
 
         // The nodal initial conditions are computed
         KRATOS_INFO_IF("MPM_Strategy", this->GetEchoLevel() > 1) << "Main Solve - InitializeSolutionStep" <<std::endl;
         mp_solving_strategy->InitializeSolutionStep();
+    }
 
+    /**
+     * @brief Operation to predict the solution
+     */
+    void Predict() override
+    {
         KRATOS_INFO_IF("MPM_Strategy", this->GetEchoLevel() > 1) << "Main Solve - Predict" <<std::endl;
         mp_solving_strategy->Predict();
+    }
 
+    /**
+     * @brief Solves the current step. This function returns true if a solution has been found, false otherwise.
+     */
+    bool SolveSolutionStep() override
+    {
         // Do solution iterations
         KRATOS_INFO_IF("MPM_Strategy", this->GetEchoLevel() > 1) << "Main Solve - SolveSolutionStep" <<std::endl;
         mp_solving_strategy->SolveSolutionStep();
 
+        return true;
```

Comment:
why not `return mp_solving_strategy->SolveSolutionStep();`?
(in case you convergence does not depend on other parameters)